featuretools/tests/synthesis/test_deep_feature_synthesis.py

```diff
@@ -1063,58 +1068,67 @@ def test_primitive_options_with_globals(es):
         # ignoring cohorts means no features are created
         assert not isinstance(f.primitive, Month)
 
-        deps = f.get_dependencies()
+        deps = f.get_dependencies(deep=True)
         entities = [d.entity.id for d in deps]
-        identities = [d for d in deps if isinstance(d, IdentityFeature)]
-        variables = [d.variable.id for d in identities]
+        variables = [d for d in deps if isinstance(d, IdentityFeature)]
         if isinstance(f.primitive, Mode):
             assert 'sessions' in entities or 'customers' in entities
-            if 'customers' in entities:
-                assert 'age' not in variables
-                assert u'région_id' not in variables
+            for identity_base in variables:
+                assert not (identity_base.entity.id == 'customers' and
+                            (identity_base.get_name() == 'age' or
+                             identity_base.get_name() == u'région_id'))
         elif isinstance(f.primitive, NumUnique):
             assert 'sessions' in entities or 'customers' in entities
-            if 'sessions' in entities:
-                assert 'device_type' in variables or variables == []
+            for identity_base in variables:
```

Comment:
should this is also check that customer age is not a base variable for NumUnique?